components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -156,24 +220,50 @@ void Nudging::initialize_impl (const RunType /* run_type */)
     src_input.finalize();
     pmid_ext.sync_to_dev();
   }
+
+  // Open the registration!
+  if(m_refine_remap) {
+    refine_remapper->registration_begins();
+  }
+
+  // To create helper fields for later; we do both hxt and ext...
   for (auto name : m_fields_nudge) {
     std::string name_ext = name + "_ext";
+    std::string name_hxt = name + "_hxt";
     // Helper fields that will temporarily store the target state, which can then
     // be used to back out a nudging tendency
+    auto grid_int_name = m_grid->name();
+    auto grid_ext_name = grid_ext->name();
+    auto grid_hxt_name = grid_hxt->name();
     auto field  = get_field_out_wrap(name);
     auto layout = field.get_header().get_identifier().get_layout();
-    create_helper_field(name,     layout,              grid_name, ps);
-    create_helper_field(name_ext, scalar3d_layout_mid, grid_name, ps);
+    create_helper_field(name,     layout,              grid_int_name, ps);
+    create_helper_field(name_ext, scalar3d_layout_mid, grid_ext_name, ps);
+    create_helper_field(name_hxt, scalar3d_layout_hid, grid_hxt_name, ps);
+    // No need to follow with hxt because we are not reading it externally
     auto field_ext = get_helper_field(name_ext);
+    if(m_refine_remap) {
+      auto field_hxt = get_helper_field(name_hxt);
+      auto field_int = get_helper_field(name);
+      refine_remapper->register_field(field_hxt, field_int);
+    }
```

Comment:
E.g., using DoNothingRemapper when remap is not necessary, here you could do:
```
auto field_int = create_helper_field(name, ...);
auto field_hxt = create_helper_field(name_hxt);
refine_remapper->register_field(field_hxt,field_int);
```
In case the remapper is the do nothing one, this call is almost a no op.